src/v2/Components/v2/ArtworkFilter/ArtworkFilterMobileActionSheet.tsx

```diff
@@ -34,6 +35,22 @@ export const ArtworkFilterMobileActionSheet: SFC<{
     contentRef.current.scrollTop = 0
   }
 
+  useEffect(() => {
+    // While mobile sheet is mounted, the effect of the user's filter selections
+    // should be merely staged until the Apply button is pressed, rather than
+    // applied immediately. Therefore…
+    //
+    // On mount, enter staged mode, and initialize a set of staged filter
+    // changes from the current filter choices.
```

Comment:
Comment pretty much says it^. I went for more verbose comments than usual here just to spell out this tricky lifecycle.